sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -831,6 +831,15 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   def table(tableName: String): DataFrame = {
     assertNoSpecifiedSchema("table")
+    val isStreamingTempView =
+      sparkSession.sessionState.catalog.getTempView(tableName).exists(_.isStreaming) ||
+        sparkSession.sessionState.catalog.getGlobalTempView(tableName).exists(_.isStreaming)
+    if (!SQLConf.get.getConf(SQLConf.LEGACY_ALLOW_READ_STREAMING_TEMP_VIEW)
+        && isStreamingTempView) {
+      throw new UnsupportedOperationException("Using `spark.table` or `spark.read.table` to " +
```

Comment:
Can we move the check into `ResolveTempViews`?

Reply:
Thanks, done in 2131507